Update release process after releasing v1.12.0-alpha.0

maelvls opened the pull request:
The most notable change is that it is now no longer necessary to create a release branch for alpha releases. The release branch gets created when cutting the first beta.

**Reasoning:** as explained by @SgtCoDFish in https://github.com/cert-manager/release/pull/109, there is no need to have periodics for the branch `release-1.12` until we have cut a beta. Since alphas "follow" the master branch, the master periodics are enough.

Thus, we don't need to immediately create the release branch anymore: the release process now tells you to create the release branch when cutting the first beta.

A pleasant consequence is that it is no longer required to "update" the release branch between alphas.

PTAL @SgtCoDFish @wallrj, I would like your feedback on my decision to not create a release branch until the first beta.

wallrj reviewed (changes requested):
I'm on the fence about delaying the creation of the release branch.
I think it further complicates the release process.
It adds a new decision for the release manager when they are updating their local repo and it adds a decision when they are choosing the parameters for the release notes tool, right?

maelvls commented on content/docs/contributing/release-process.md:
```diff
@@ -347,15 +358,15 @@ page if a step is missing or if it is outdated.
 
         <div className="pageinfo pageinfo-info"><p>
-        Follow the <code>cmrel stage</code> build: https://console.cloud.google.com/cloud-build/builds/7641734d-fc3c-42e7-9e4c-85bfc4d1d547?project=1021342095237
+        Follow the <code>cmrel makestage</code> build: https://console.cloud.google.com/cloud-build/builds/7641734d-fc3c-42e7-9e4c-85bfc4d1d547?project=1021342095237
```
I changed to `cmrel makestage` because the above command changed from `cmrel stage` to `cmrel makestage` a few months ago.

maelvls commented:
> [Delaying the creation of the release branch] adds a new decision for the release manager when they are updating their local repo.

That's a good point. I added the section "Place yourself on the correct branch" with the right instructions. I named it "Place yourself on the correct branch" because it isn't necessary to "update" the branches anymore.

> It adds a decision when they are choosing the parameters for the release notes tool.

To make sure the decision is straight and easy, I updated the "example" table in the PR to reflect the change. Both the initial alpha and subsequent alpha now happen on the `master` branch:

| Variable          | Example 1        | Example 2        |
| ----------------- | ---------------- | ---------------- |
|                   |                  |                  |
|                   | initial alpha    | subsequent alpha |
| `RELEASE_VERSION` | `v1.3.0-alpha.0` | `v1.3.0-alpha.1` |
| `START_TAG`\*     | `v1.2.0`         | `v1.3.0-alpha.0` |
| `END_REV`         | `master`         | `master`         |
| `BRANCH`          | `master`         | `master`         |

Do you think we can proceed with this new release process for cert-manager 1.12, and revert back if unexpected things happen?

wallrj reviewed (approved):
Thanks. Yes, it's a good idea to try this modified process. 

/lgtm
